src/devices/Sgp30/Sgp30.cs

```diff
@@ -0,0 +1,279 @@
+            byte[] resultArray = resultBuffer.ToArray();
+            ushort result = (ushort)(resultArray[1] << 8 | resultArray[2]);
+            byte checksum = CalculateChecksum(result);
+
+            if (checksum != resultArray[3])
+            {
+                throw new ChecksumFailedException();
+            }
+
+            return result;
+        }
+
+        /// <summary>
+        /// Initialises the SGP30 for measurement. NOTE: This call will block for up to 20 seconds.
+        /// </summary>
+        /// <returns>Initial SGP30 measurement result.</returns>
+        public Sgp30Measurement InitialiseMeasurement()
+        {
+            _i2cDevice.Write(new ReadOnlySpan<byte>(new byte[] { (byte)((SGP30_INITIALISE_AIR_QUALITY & 0xFF00) >> 8), (byte)(SGP30_INITIALISE_AIR_QUALITY & 0x00FF) }));
+            Thread.Sleep(10);
+
+            Sgp30Measurement result = new Sgp30Measurement
+            {
+                Tvoc = 0x0000,
+                Eco2 = 0x0000
+            };
+
+            for (int i = 0; i < 20; i++)
+            {
+                result = GetMeasurement();
+
+                if (result.Tvoc != 0 || result.Eco2 != 400)
+                {
+                    // If either eCO2 != 400ppm or TVOC != 0ppb, the device has initialised, don't wait for the
+                    // remaining initialisation measurements to complete.
+                    break;
+                }
+
+                Thread.Sleep(1000);
+            }
+
+            return result;
+        }
+
+        /// <summary>
+        /// Get the current eCO2 and TVOC measurements from the SGP30. NOTE: This should be called at least once per
+        /// second as per the device datasheet in order to maintain accuracy of the readings. If several seconds have
+        /// passed, call <see cref="InitialiseMeasurement" />.
+        /// </summary>
+        /// <returns>SGP30 measurement result.</returns>
+        public Sgp30Measurement GetMeasurement()
+        {
+            return GetData(SGP30_MEASURE_AIR_QUALITY);
+        }
+
+        /// <summary>
+        /// Get the raw signal data without on-device processing to TVOC ppb / eCO2 ppm values.
+        /// </summary>
+        /// <returns>Raw sensor data.</returns>
+        public Sgp30Measurement GetRawSignalData()
+        {
+            return GetData(SGP30_MEASURE_RAW_SIGNALS);
+        }
+
+        /// <summary>
+        /// Get device baseline measurement data.
+        /// </summary>
+        /// <returns>Measured baseline data.</returns>
+        public Sgp30Measurement GetBaseline()
+        {
+            return GetData(SGP30_GET_BASELINE);
+        }
+
+        /// <summary>
+        /// Generalised data read for getting Measurement, Raw Measurement and Baseline data.
+        /// </summary>
+        /// <param name="command">Command to send to the SGP30.</param>
+        /// <returns>Resulting data returned from SGP30.</returns>
+        private Sgp30Measurement GetData(ushort command)
```

Comment:
As mentionned before, would really mke sense to have an internal enum with all the possible commands.